opentelemetry-api/src/opentelemetry/trace/status.py

```diff
@@ -0,0 +1,184 @@
+    for the operation's execution.
+
+    For example, the directory to be deleted is non-empty, an rmdir operation
+    is applied to a non-directory, etc. Service implementors can use the
+    following guidelines to decide between FAILED_PRECONDITION, ABORTED, and
+    UNAVAILABLE:
+
+        (a) Use UNAVAILABLE if the client can retry just the failing call.
+        (b) Use ABORTED if the client should retry at a higher level (e.g.,
+            when a client-specified test-and-set fails, indicating the client
+            should restart a read-modify-write sequence).
+        (c) Use FAILED_PRECONDITION if the client should not retry until the
+            system state has been explicitly fixed.
+
+    E.g., if an "rmdir" fails because the directory is non-empty,
+    FAILED_PRECONDITION should be returned since the client should not retry
+    unless the files are deleted from the directory.
+    """
+
+    ABORTED = 10
+    """The operation was aborted, typically due to a concurrency issue such as a
+    sequencer check failure or transaction abort.
+
+    See the guidelines above for deciding between FAILED_PRECONDITION, ABORTED,
+    and UNAVAILABLE.
+    """
+
+    OUT_OF_RANGE = 11
+    """The operation was attempted past the valid range.
+
+    E.g., seeking or reading past end-of-file. Unlike INVALID_ARGUMENT, this
+    error indicates a problem that may be fixed if the system state changes.
+    For example, a 32-bit file system will generate INVALID_ARGUMENT if asked
+    to read at an offset that is not in the range [0,2^32-1],but it will
+    generate OUT_OF_RANGE if asked to read from an offset past the current file
+    size. There is a fair bit of overlap between FAILED_PRECONDITION and
+    OUT_OF_RANGE. We recommend using OUT_OF_RANGE (the more specific error)
+    when it applies so that callers who are iterating through a space can
+    easily look for an OUT_OF_RANGE error to detect when they are done.
+    """
+
+    UNIMPLEMENTED = 12
+    """The operation is not implemented or is not supported/enabled in this
+    service.
+    """
+
+    INTERNAL = 13
+    """Internal errors.
+
+    This means that some invariants expected by the underlying system have been
+    broken. This error code is reserved for serious errors.
+    """
+
+    UNAVAILABLE = 14
+    """The service is currently unavailable.
+
+    This is most likely a transient condition, which can be corrected by
+    retrying with a backoff.  Note that it is not always safe to retry
+    non-idempotent operations.
+    """
+
+    DATA_LOSS = 15
+    """Unrecoverable data loss or corruption."""
+
+    UNAUTHENTICATED = 16
+    """The request does not have valid authentication credentials for the
+    operation.
+    """
+
+
+class Status:
+    """Represents the status of a finished Span.
+
+    Args:
+        canonical_code: The canonical status code that describes the result status of the operation.
```

Comment:
```suggestion
        canonical_code: The canonical status code that describes the result
            status of the operation.
```

I agree we need a linter for this.